thinc/loss.py

```diff
@@ -143,6 +146,26 @@ def get_grad(self, guesses: Floats2d, truths: IntsOrFloatsOrStrs) -> Floats2d:
             raise ValueError(err)
         difference = guesses - target
         difference *= mask
+        # Weight samples proportional to the marginals.
+        # For each sample the target labels are normalized
+        # to sum to 1 and the normalized labels are used
+        # to compute the weighted average of class_weights
+        if self.class_weights is not None:
+            if target.shape[1] != len(self.class_weights):
+                raise ValueError(
+                    "The number of classes in the "
+                    "target and in class weights "
+                    "has to be equal, but found "
+                    f"{target.shape[1]} and "
+                    f"{len(self.class_weights)}"
+                )
+            if not is_xp_array(self.class_weights):
+                self.class_weights = xp.asarray(self.class_weights)
```

Comment:
Maybe we should do this in the constructor, so that it is only done once?

Reply:
Oh, it is only done once. But I guess it'd still make sense to do this in the constructor? Simplifies the invariants within the class.